docs/src/transforms.md

```diff
@@ -0,0 +1,96 @@
+# [Transforms](@id about-transforms)
+
+A `Transform` defines a transformation of data, for example scaling, periodic functions, linear combination, one-hot encoding, etc.
+
+```@meta
+DocTestSetup = quote
+    using DataFrames
+    using Dates
+    using FeatureTransforms
+end
+```
+
+Usually, a `Transform` has one or more parameters. For example, we can define a squaring operation (i.e. raise to the power of 2):
+
+```jldoctest transforms
+julia> p = Power(2);
+```
+
+This transformation can be applied to data `x` as follows:
+
+```jldoctest transforms
+julia> x = [1.0, 2.0, 3.0];
+
+julia> FeatureTransforms.apply(x, p)
+3-element Array{Float64,1}:
+ 1.0
+ 4.0
+ 9.0
+```
+
+## Applying a `Transform`
+
+There are three main ways to apply a `Transform` - suppose it is called `t`:
+
+* `Transforms.apply(data, t; kwargs...)` is non-mutating, returning the transformed data without modifying the original data.
+* `t(data; kwargs...)` is equivalent to `Transforms.apply(data, t; kwargs...)`
+* `Transforms.apply!(data, t; kwargs...)` is mutating, returning the modified `data` with the same type.
+
+A single `Transform` can be applied to different data types and in different ways. The two main data types supported are `AbstractArray`s and [`Table`s](https://github.com/JuliaData/Tables.jl).
+
+### `AbstractArray` data
+
+For `AbstractArray` data, some `Transform`s support a `dims` keyword argument in their `apply` methods. This will apply the `Transform` to slices of the array along dimensions determined by `dims`. For example, given a `Matrix`, `dims=1` applies to each column, and `dims=2` applies
+to each row. This convention is similar to `Statistics.mean(M; dims=2)` returning the mean of each row in matrix `M`.
```

Comment:
I feel like we need to tell 3 different things here and we may want to split them up a bit better

1. How a vanilla `Transform` is applied to an array (e.g. Power)
2. How `inds` works so one can only apply to certain slices 
3. How `dims` works - demonstrated by a slightly more complex transform (e.g. `LinearCombination`).

> This convention is similar to `Statistics.mean(M; dims=2)` returning the mean of each row in matrix `M`.

I think this is helpful to mention (as it will immediately clarify the output to users) but we should use it as a complement to explaining how we use `mapslices` under the hood.

Reply:
Yeah, clear headed sections for each thing would be kinder to the reader.

> I think this is helpful to mention (as it will immediately clarify the output to users) but we should use it as a complement to explaining how we use mapslices under the hood.

Not sure what you mean - you want to explain in the doc how we use `mapslices` under the hood?

Reply:
>Not sure what you mean - you want to explain in the doc how we use mapslices under the hood?

As in, we should say something like 

> The `dims` argument uses the convention used by `mapslices`, which is called behind the scenes when applying transformations to slices of data. In practice, this means that users can expect the `dims` keyword to behave exactly as `mean(A; dims=d)` would; the transformation will be applied to the elements along the dimension `d` and, for operations like `mean` or `sum`, reduce across this dimension.